src/internal_events/open.rs

```diff
@@ -0,0 +1,129 @@
+use super::InternalEvent;
+use metrics::gauge;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+pub type OpenTokenDyn = OpenToken<Box<dyn Fn(usize) + 'static + Send>>;
+
+#[derive(Debug)]
+pub struct ConnectionOpen {
+    pub count: usize,
+}
+
+impl InternalEvent for ConnectionOpen {
+    fn emit_metrics(&self) {
+        gauge!("open_connections", self.count as f64);
+    }
+}
+
+#[derive(Clone)]
+pub struct OpenGauge {
+    gauge: Arc<AtomicUsize>,
+}
+
+impl OpenGauge {
+    pub fn new() -> Self {
+        OpenGauge {
+            gauge: Arc::default(),
+        }
+    }
+
+    /// Increments and emits value once created.
+    /// Decrements and emits value once dropped.
+    pub fn open<E: Fn(usize)>(self, emitter: E) -> OpenToken<E> {
+        gauge_add(&self.gauge, 1, &emitter);
+        OpenToken {
+            gauge: self.gauge,
+            emitter,
+        }
+    }
+}
+
+impl Default for OpenGauge {
```

Comment:
Can be defined through `#[derive(Clone, Default)]`. It's strange that clippy did not catch `new` without arguments :thinking: